docs/7.Scan Examples/OpenAPI.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: default
+published: true
+title: OpenAPI configuration scanning
+nav_order: 20
+---
+
+# OpenAPI configuration scanning
+Checkov supports the evaluation of policies on your OpenAPI files.
+When using checkov to scan a directory that contains a OpenAPI manifests it will validate if the file is compliant with OpenAPI best practices such as securityDefinitions and security requirement are well-defined, and more.  
+
+Full list of OpenAPI policies checks can be found [here](https://www.checkov.io/5.Policy%20Index/openapi.html).
+
+### Example misconfigured OpenAPI
+
+```yaml
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "example",
+    "version": "1.0.0"
+  },
+  "paths": {
+    "/": {
+      "get": {
+        "operationId": "example",
+        "summary": "example",
+        "responses": {
+          "200": {
+            "description": "200 response"
+          }
+        },
+        "parameters": [
+          {
+            "name": "limit2",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "type": "object"
+            }
+          }
+        ],
+        "security": [
+          {
+            "api_key": []
+          }
+        ]
+      }
+    }
+  },
+  "securityDefinitions": {
+    "petstore_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "http://swagger.io/api/oauth/dialog",
+      "flow": "implicit",
+      "scopes": {
+        "write:pets": "write",
+        "read:pets": "read"
+      }
+    }
+  }
+}
+
+```
+### Running in CLI
+
+```bash
+checkov -d . --framework openapi
+```
+
+### Example output
+```bash
+ 
+       _               _              
+   ___| |__   ___  ___| | _______   __
+  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
+ | (__| | | |  __/ (__|   < (_) \ V / 
+  \___|_| |_|\___|\___|_|\_\___/ \_/  
+                                      
+By bridgecrew.io | version: 2.0.1046 
+Update available 2.0.1046 -> 2.0.1050
+Run pip3 install -U checkov to update 
+
+
+openapi scan results:
+
+Passed checks: 4, Failed checks: 2, Skipped checks: 0
+
+Check: CKV_OPENAPI_2: "Ensure that if the security scheme is not of type 'oauth2', the array value must be empty"
+	PASSED for resource: security
+	File: /openapi.yaml:2-47
+Check: CKV_OPENAPI_5: "Ensure that security operations is not empty."
+	PASSED for resource: security
+	File: /openapi.yaml:2-47
+Check: CKV_OPENAPI_1: "Ensure that securityDefinitions is defined and not empty."
+	PASSED for resource: securityDefinitions
+	File: /../aya/openapi.yaml:36-46
+Check: CKV_OPENAPI_3: "Ensure that security schemes don't allow cleartext credentials over unencrypted channel"
+	PASSED for resource: components
+	File: /openapi.yaml:2-47
+Check: CKV_OPENAPI_6: "Ensure that security requirement defined in securityDefinitions."
+	FAILED for resource: security
+	File: /openapi.yaml:29-31
+
+		29 |             "api_key": []
+		30 |           }
+		31 |         ]
```

Comment:
this somehow looks weird, is the output correct? maybe the check needs to be adjusted?